py_ecc/fields/field_elements.py

```diff
@@ -0,0 +1,366 @@
+            )
+
+        return type(self)([x - y for x, y in zip(self.coeffs, other.coeffs)])
+
+    def __mul__(self, other: Union[int, "FQ", "FQP"]) -> "FQP":
+        if isinstance(other, int_types_or_FQ):
+            return type(self)([c * other for c in self.coeffs])
+        elif isinstance(other, FQP):
+            b = [self.FQP_corresponding_FQ_class(0) for i in range(self.degree * 2 - 1)]
+            for i in range(self.degree):
+                for j in range(self.degree):
+                    b[i + j] += self.coeffs[i] * other.coeffs[j]
+            while len(b) > self.degree:
+                exp, top = len(b) - self.degree - 1, b.pop()
+                for i in range(self.degree):
+                    b[exp + i] -= top * self.FQP_corresponding_FQ_class(self.modulus_coeffs[i])
+            return type(self)(b)
+        else:
+            raise TypeError(
+                "Expected an int or FQ object or FQP object, but got object of type {}"
+                .format(type(other))
+            )
+
+    def __rmul__(self, other: Union[int, "FQ", "FQP"]) -> "FQP":
+        return self * other
+
+    def __div__(self, other: Union[int, "FQ", "FQP"]) -> "FQP":
+        if isinstance(other, int_types_or_FQ):
+            return type(self)([c / other for c in self.coeffs])
+        elif isinstance(other, FQP):
+            return self * other.inv()
+        else:
+            raise TypeError(
+                "Expected an int or FQ object or FQP object, but got object of type {}"
+                .format(type(other))
+            )
+
+    def __truediv__(self, other: Union[int, "FQ", "FQP"]) -> "FQP":
+        return self.__div__(other)
+
+    def __pow__(self, other: int) -> "FQP":
+        if other == 0:
+            return type(self)([1] + [0] * (self.degree - 1))
+        elif other == 1:
+            return type(self)(self.coeffs)
+        elif other % 2 == 0:
+            return (self * self) ** (other // 2)
+        else:
+            return ((self * self) ** int(other // 2)) * self
+
+    # Extended euclidean algorithm used to find the modular inverse
+    def inv(self) -> "FQP":
+        lm, hm = (
+            [1] + [0] * self.degree,
+            [0] * (self.degree + 1),
+        )
+        low, high = (
+            # Ignore mypy yelling about the inner types for  the tuples being incompatible
+            cast(List[IntOrFQ], list(self.coeffs + (0,))),  # type: ignore
+            cast(List[IntOrFQ], list(self.modulus_coeffs + (1,))),  # type: ignore
+        )
+        while deg(low):
+            r = cast(List[IntOrFQ], list(poly_rounded_div(high, low)))
+            r += [0] * (self.degree + 1 - len(r))
+            nm = [x for x in hm]
+            new = [x for x in high]
+            if len(set(
+                [len(lm), len(hm), len(low), len(high), len(nm), len(new), self.degree + 1]
+            )) != 1:
+                raise Exception("Mismatch between the lengths of lm, hm, low, high, nm, new")
```

Comment:
I know it is more work but it would be ideal to actually include which one was the mismatch.

Reply:
On this.

Reply:
Also please note that, we could display an error if only one of them all, don't match with the rest. Assuming that would be the scenario, I could cook something up.